### PR TITLE
Add sparse color attachments operation tests

### PR DESCRIPTION
**Pull request by shrekshao**
Following #1149 (to use new webgpu types)

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts [hunk 1]
```diff
@@ -1,75 +1,209 @@
...
+      .combine('attachmentCount', [2, 3, 4])
+      .expand('emptyAttachmentId', p => range(p.attachmentCount, i => i))
+  )
+  .fn(async t => {
+    const { format, attachmentCount, emptyAttachmentId } = t.params;
+    const componentCount = kTexelRepresentationInfo[format].componentOrder.length;
+    const info = kTextureFormatInfo[format];
+    await t.selectDeviceOrSkipTestCase(info.feature);
+
+    const writeValue = { R: 1, G: 1, B: 1, A: 1 };
```
Comment (kainino0x) [inline src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts:142]: Probably want a different writeValue for each attachment so we can make sure they didn't get mixed up somehow

Comment (shrekshao) [reply src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts]: Fixed (also set different value for sample type = float/int)

File: src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts [hunk 2]
```diff
@@ -139,7 +139,8 @@ g.test('color,attachments')
     const info = kTextureFormatInfo[format];
     await t.selectDeviceOrSkipTestCase(info.feature);
 
-    const writeValue = { R: 0.125, G: 0.25, B: 0.5, A: 1 };
+    const writeValue =
+      info.sampleType === 'float' ? { R: 0.2, G: 0.6, B: 0.8, A: 1 } : { R: 2, G: 4, B: 8, A: 16 };
```
Comment (kainino0x) [inline src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts:143]: I meant a for each different attachment index, not for each component (although that is also good). I don't know how easy that would be to do though.

Comment (shrekshao) [reply src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts:143]: Was making an attempt to use the new `bitsToNumber`/`numberToBits` conversion to take care of float/unorm precision issue for different value but doesn't fully work for 16bit float. https://github.com/gpuweb/cts/commit/cbbc37893d5c123b8f31043a785bf81e29f20de0#diff-1bb578801feab213197f6db552e57d91911c1e537576b8c3e3179d09ec59c5ccR182

Seems a good place for some more texture expect helper (within a threshold). Will try on later

Comment (kainino0x) [reply src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts:143]: The texture helpers I wrote recently have fancy thresholding options, and we want to transition the old texture helper functions onto that ideally. #1121

**Pull request merged**